iceoryx_utils/include/iceoryx_utils/cxx/unique_ptr.hpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_UTILS_CXX_UNIQUE_PTR_HPP
+#define IOX_UTILS_CXX_UNIQUE_PTR_HPP
+
+#include "iceoryx_utils/cxx/function_ref.hpp"
+#include <functional>
+
+namespace iox
+{
+namespace cxx
+{
+///
+/// @todo document how it differs to STL
+///
+template <typename T>
+class unique_ptr
+{
+  public:
+    using ptr_t = T*;
+
+    unique_ptr() = delete;
```

Comment:
Why is this ctor deleted. It does in the end the same like the ctor below: `unique_ptr(std::nullptr_t) noexcept;` creating an empty unique ptr.

Reply:
Yes but I would rather people are explicit when initializing an empty pointer.
e.g.
Allow `unique_ptr<SomeStruct> ptr = nullptr` but not `unique_ptr<SomeStruct> ptr = unique_ptr<SomeStruct>()`